### ISSUE TITLE
CopyBentleyStaticResourcesPlugin not compatible with "clean-webpack-plugin"

### ISSUE THREAD

**Issue by alarkbentley** [imodeljs/imodeljs#406]
When using CopyBentleyStaticResourcesPlugin together with the external plugin [clean-webpack-plugin](https://github.com/johnagan/clean-webpack-plugin), the static resources are never included in the build output.

CopyBentleyStaticResourcesPlugin unpacks the static resources to build directory before "clean-webpack-plugin" cleans the output, thus the static resources are removed.

Reason is that CopyBentleyStaticResourcesPlugin is executed in "compiler.hooks.beforeRun" while clean-webpack-plugin is done in "compiler.hooks.emit". Can we copy resources in a later build step?

**Comment by alarkbentley** [comment]
A potential solution is to use "compilation.hooks.processAssets" instead of "compilation.hooks.beforeRun". This is how [copy-webpack-plugin](https://webpack.js.org/plugins/copy-webpack-plugin/) does it and it works with clean-webpack-plugin.

**Comment by calebmshafer** [comment]
@alarkbentley that suggestion makes sense to me. @wgoehrig any concerns about moving the copy plugin to a step time in the webpack workflow?

**Comment by aruniverse** [comment]
@alarkbentley, can we close this issue now? Was your issue resolved with the CopyStaticAssetsPlugin?

**Comment by aruniverse** [comment]
> I haven't been able to test properly because of the linux issue, has it been backported to 2.16.x yet?

Yes it was backported and released in 2.16.2 last week. Please test it out and let us know if it works; and if we can close this issue.

**Comment by alarkbentley** [comment]
Tested now with 2.16.2 and it works!
